xbmc/pvr/guilib/PVRGUIActionListener.cpp

```diff
@@ -37,7 +39,10 @@ namespace PVR
 
 CPVRGUIActionListener::CPVRGUIActionListener()
 {
-  g_application.RegisterActionListener(this);
+  auto& components = CServiceBroker::GetAppComponents();
+  const auto appListener = components.GetComponent<CApplicationActionListeners>();
+  if (appListen)
```

Comment:
should an error be logged (or a logic_error be thrown) if this component is not available? The app will not work correctly if the listener cannot be registered.

Reply:
i'm not sure how to handle things tbh. 

as is, in the actual code, there's really no way we can be here without the listener being available, since the application is a global initialized before even main() is run, and components are registered in the ctor, and removed in the dtor. however, not checking the pointers will get static analyzers panties in a bundle. it's more an assert
than anything else. this will pop up all over with these changes, so we should give it some thought..

i was contemplating a compile time construct instead of a run time one since then we can static assert on such things. however, i decided not to go that way as that means no state in the components which is also rather limiting.

Reply:
`throw std::logic_error`?

Reply:
it will have to be done in 100's of locations.

Reply:
You will touch all these 100's of locations anyway, right? It would just be one more line to add... but no hard feelings, except silently ignoring error makes me shiver. I know this is a difficult topic, because the anti-pattern to silently ignore errors even if we cannot recover from those, can be found all over the place in Lodi code base.

Famous example: Request a dialog instance, which has to be available for the app to work properly, but just skip what should have be done with dialog in case window mgr does not return the requested instance. Effect for the user, at best nothing happens, ever they pushed a button or the like.

Reply:
But in no way I will block this PR. I feel we will not discuss this to the end here. ;-)

Reply:
i am in no way going to silently ignore the errors, as i said it's mostly a matter of how to handle non-errors. i'm fine with using a throw, but then i'd rather throw if an nonexistent component is grabbed from ComponentContainer. we get the same effect, and i do not have litter if's everywere.